aodncore/pipeline/handlerbase.py

```diff
@@ -138,39 +138,123 @@
 ----------
 
 A handler instance contains a number of attributes which control or modify the behaviour of the handler. The attributes
-are typically set from the **params** key of the watch configuration, or from the ``__init__`` method of a handler
-subclass (e.g. when writing tests).
+are typically set from the **params** key of the watch configuration, or as initialisation parameters to ``__init__``
+method of a handler subclass (e.g. when writing tests).
 
 Class parameters
 ~~~~~~~~~~~~~~~~
 
-The following class parameters are also assigned to attributes of the same name, as a convenience.
+The class parameters are also assigned to instance attributes of the same name, as a convenience.
 
-For example, a handler instantiated with any of these class parameters may also access them from the class instance as
-follows::
+A handler instantiated with any of these class parameters may also access them from the class instance::
```

Comment:
For some reason I find the last two statements confusing, even though I'm pretty sure I understand what you're talking about. Above you already explained that "instance attributes" are set from "initialisation parameters" or "params" in the watch config. Here you use the term "class parameters", but I don't think you're saying anything new.
I guess what I'm saying is try to use fewer similar-but-distinct terms.
The examples below certainly help!

Reply:
It's not just you, it's confusing for me to read it back!

I agree, the terminology needs to be simple and consistent, and isn't quite there yet.

What I'm trying to find the terms for are the two distinct concepts of *class arguments/parameters*, i.e. those passed when instantiating the class (the __init__ arguments), and the instance *attributes*. e.g.

```python

class MyHandler(HandlerBase):
    # 'this' and 'that' are class parameters/arguments
    def __init__(self, this, that):
        # 'self.this' and 'self.that' are instance attributes
        self.this = this
        self.that = that

```

I based it on this doco here, which I thought was a nice simple idea, to make the arguments and attributes consistent: http://docs.celeryproject.org/en/latest/userguide/tasks.html#list-of-options

For example, so that a user knows that they can always access the "check_params" they provided to the class via "self.check_params".

Maybe just need to settle on an accepted term for both things, and add a note to explain in simple terms what they mean?

Reply:
Yes. I think all you need to do is explain that _class parameters_ are the arguments of the `__init__` function, which is called when a new handler instance is created. In turn, that function assigns these parameters to corresponding _instance attributes_, which can be accessed within handler methods as `self.<attribute>`. How's that?
The brief example in your reply above might help to make this clear.